### PR TITLE
feat: Align MetricPipeline conditions with Kubernetes API conventions

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request (what was done and why):

- Refactor MetricPipeline status conditions to static map-like slice, replacing self-made Condtions with metav1.Conditions
- Adapt aggregated status in TelemetryCR
- Adjust the Busola extensions
- Adjust the docs

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/601

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [x] The feature is e2e-tested

### CODE REVIEW

File: internal/reconciler/metricpipeline/status.go [hunk 1]
```diff
@@ -15,139 +16,97 @@ import (
...
+	if !lockAcquired {
+		status = metav1.ConditionFalse
+		reason = conditions.ReasonWaitingForLock
 	}
 
-	pendingAgent := telemetryv1alpha1.NewMetricPipelineCondition(conditions.ReasonMetricAgentDaemonSetNotReady, telemetryv1alpha1.MetricPipelinePending)
-	if pipeline.Status.HasCondition(telemetryv1alpha1.MetricPipelineRunning) {
-		log.V(1).Info(fmt.Sprintf("Updating the status of %s to %s. Resetting previous conditions", pipeline.Name, pendingAgent.Type))
-		pipeline.Status.Conditions = []telemetryv1alpha1.MetricPipelineCondition{}
+	referencesNonExistentSecret := secretref.ReferencesNonExistentSecret(ctx, r.Client, pipeline)
```
Comment (chrkl) [inline internal/reconciler/metricpipeline/status.go:95]: I'm wondering if we should change the order here. My first intuition is that the pipeline limit is the "hardest" condition and if that is reached, we should indicate that we entirely ignore the pipeline.

**Review by NHingerl — DISMISSED** [review]

File: internal/testutils/metric_pipeline_builder.go [hunk 2]
```diff
@@ -186,18 +185,29 @@ func (b *MetricPipelineBuilder) Build() telemetryv1alpha1.MetricPipeline {
 						Basic: &telemetryv1alpha1.BasicAuthOptions{
 							User: telemetryv1alpha1.ValueType{
 								Value: b.basicAuthUser,
+								ValueFrom: &telemetryv1alpha1.ValueFromSource{
```
Comment (chrkl) [inline internal/testutils/metric_pipeline_builder.go:188]: Does this mean we did not support basic auth with secret references before?

Comment (skhalash) [reply internal/testutils/metric_pipeline_builder.go:188]: In the testutils, yes

File: internal/reconciler/metricpipeline/status.go (hunk 1 above)
Comment (skhalash) [reply internal/reconciler/metricpipeline/status.go:95]: Yeah, I agree. I will change it and adapt the tests

**Review by NHingerl — APPROVED** [review]

**Pull request merged**